client/CPreGame.cpp

```diff
@@ -3167,37 +3151,35 @@ CMultiMode::CMultiMode()
 	bar = new CGStatusBar(new CPicture(Rect(7, 465, 440, 18), 0));//226, 472
 	txt = new CTextInput(Rect(19, 436, 334, 16), *bg);
 	txt->setText(settings["general"]["playerName"].String()); //Player
+	txt->cb += std::bind(&CMultiMode::onNameChange, this, _1);
 
-	btns[0] = new CButton(Point(373, 78),        "MUBHOT.DEF",  CGI->generaltexth->zelp[266], std::bind(&CMultiMode::openHotseat, this));
-	btns[1] = new CButton(Point(373, 78 + 57*1), "MUBHOST.DEF", CButton::tooltip("Host TCP/IP game", ""), std::bind(&CMultiMode::hostTCP, this));
-	btns[2] = new CButton(Point(373, 78 + 57*2), "MUBJOIN.DEF", CButton::tooltip("Join TCP/IP game", ""), std::bind(&CMultiMode::joinTCP, this));
-	btns[6] = new CButton(Point(373, 424),       "MUBCANC.DEF", CGI->generaltexth->zelp[288], [&](){ GH.popIntTotally(this);}, SDLK_ESCAPE);
-}
+	btns[0] = new CButton(Point(373, 78), "MUBHOT.DEF", CGI->generaltexth->zelp[266], std::bind(&CMultiMode::hostTCP, this));
+	btns[1] = new CButton(Point(373, 78 + 57 * 1), "MUBHOST.DEF", CButton::tooltip("Host TCP/IP game", ""), std::bind(&CMultiMode::hostTCP, this));
+	btns[2] = new CButton(Point(373, 78 + 57 * 2), "MUBJOIN.DEF", CButton::tooltip("Join TCP/IP game", ""), std::bind(&CMultiMode::joinTCP, this));
+	btns[6] = new CButton(Point(373, 424), "MUBCANC.DEF", CGI->generaltexth->zelp[288], [&]() { GH.popIntTotally(this);}, SDLK_ESCAPE);
 
-void CMultiMode::openHotseat()
-{
-	GH.pushInt(new CHotSeatPlayers(txt->text));
 }
 
 void CMultiMode::hostTCP()
 {
-	Settings name = settings.write["general"]["playerName"];
-	name->String() = txt->text;
 	GH.popIntTotally(this);
-	if(settings["session"]["donotstartserver"].Bool())
-		GH.pushInt(new CSimpleJoinScreen(CMenuScreen::MULTI_NETWORK_HOST));
-	else
-		GH.pushInt(new CSelectionScreen(CMenuScreen::newGame, CMenuScreen::MULTI_NETWORK_HOST));
+	GH.pushInt(new CMultiPlayers(settings["general"]["playerName"].String(), CMenuScreen::MULTI_NETWORK_HOST, state));
 }
 
 void CMultiMode::joinTCP()
+{
+	GH.popIntTotally(this);
+	GH.pushInt(new CMultiPlayers(settings["general"]["playerName"].String(), CMenuScreen::MULTI_NETWORK_GUEST, state));

```

Comment:
How this even works? You accessing `state` field after destruction.

Reply:
Oops, but somehow it does work for sure.